code/controllers/subsystem/title.dm

```diff
@@ -35,22 +35,27 @@ SUBSYSTEM_DEF(title)
 	//Calculate the screen size
 	var/width = round(single_frame.Width() / world.icon_size)
 	var/height = round(single_frame.Height() / world.icon_size)
-	lobby_screen_size = "[width]x[height]"
-
-	//Update the new player start (views are centered)
-	var/new_player_x = splash_turf.x + FLOOR(width / 2, 1)
-	var/new_player_y = splash_turf.y + FLOOR(height / 2, 1)
-	newplayer_start_loc = locate(new_player_x, new_player_y, splash_turf.z)
-	for(var/atom/movable/new_player_start in GLOB.newplayer_start)
-		new_player_start.forceMove(newplayer_start_loc)
-
-	//Update fast joiners
-	for (var/mob/dead/new_player/fast_joiner in GLOB.new_player_list)
-		if(isnull(fast_joiner.client?.view_size))
-			fast_joiner.client?.change_view(getScreenSize(fast_joiner))
-		else
-			fast_joiner.client?.view_size.resetToDefault(getScreenSize(fast_joiner))
-		fast_joiner.forceMove(newplayer_start_loc)
+
+	// Set the max allowed size of a lobby screen to the width of the z-level
+	if (width < world.maxx)
+		lobby_screen_size = "[width]x[height]"
+
+		//Update the new player start (views are centered)
+		var/new_player_x = splash_turf.x + FLOOR(width / 2, 1)
+		var/new_player_y = splash_turf.y + FLOOR(height / 2, 1)
+		newplayer_start_loc = locate(new_player_x, new_player_y, splash_turf.z)
+		GLOB.newplayer_start.Cut()
+		GLOB.newplayer_start += newplayer_start_loc
+
+		//Update fast joiners
+		for (var/mob/dead/new_player/fast_joiner in GLOB.new_player_list)
+			if(isnull(fast_joiner.client?.view_size))
+				fast_joiner.client?.change_view(getScreenSize(fast_joiner))
+			else
+				fast_joiner.client?.view_size.resetToDefault(getScreenSize(fast_joiner))
+			fast_joiner.forceMove(newplayer_start_loc)
+	else
+		stack_trace("Title screen exception: The title screen [file_path] is not properly formatted as a dmi and cannot be properly understood by Byond. Please ensure that the image actually has the correct extension (.dmi file is actually a .gif).")
```

Comment:
I'm trying to say that .gif files shouldn't have the .dmi extension. If its going to be a .dmi file, then it should be a .dmi file in format otherwise byond messes up with image related procs.